src/mono/mono/metadata/memory-manager.c

```diff
@@ -66,7 +66,7 @@ lock_free_mempool_alloc0 (LockFreeMempool *mp, guint size)
 
 	// FIXME: Free the allocator
 
-	size = ALIGN_TO (size, 8);
+	size = (guint)ALIGN_TO (size, 8);
```

Comment:
So my thinking here has been: if it's of the form `x = ALIGN_TO (x, N)` then I just write it as `x = (typeof_x) ALIGN_TO (x, N)` because there's nothing to check.

Reply:
OK, I see, as long as that rule applies, storing back to the same type, then the issue is triggered just due to the internals of ALIGN_TO moving over gssize and back to type. Totally agree with that heuristic, so then we just need to make sure we detect cases triggering data truncation where the variable getting converted value is of different type and mitigate them accordingly.